libraries/core/classes/AppCommand/index.js

```diff
@@ -1,169 +1,171 @@
-    if (params) {
-      this.command.p = params;
-    }
-
-    if (hasSGJavaScriptBridge()) {
-      try {
-        if ('dispatchCommandsForVersion' in SGJavascriptBridge) {
-          SGJavascriptBridge.dispatchCommandsForVersion([this.command], this.libVersion);
-        } else {
-          SGJavascriptBridge.dispatchCommandsStringForVersion(
-            JSON.stringify([this.command]),
-            this.libVersion
-          );
-        }
-
-        this.logCommand();
-      } catch (exception) {
-        logger.error(exception);
-      }
-    } else {
-      this.logCommand();
-
-      const devServerCommands = [
-        'sendPipelineRequest',
-        'sendHttpRequest',
-        'sendDataRequest',
-        'getWebStorageEntry',
-      ];
+  logCommand() {
+    if (this.log && !this.commandsWithoutLog.includes(this.name)) {
+      const title = `AppCommand %c${this.name}`;
 
-      // Only commands which represent a request are sent to the development server.
-      if (devServerCommands.includes(this.command.c)) {
-        this.sendDevCommand();
-      }
+      logGroup(title, this.params || {}, '#8e44ad');
     }
+
+    return this;
   }
 
   /**
-   * Logs the command to the console.
+   * Creates the command object which will be dispatched through the JavaScript bridge.
+   * @private
+   * @return {Object|null}
    */
-  logCommand() {
-    if (
-      !this.log ||
-      this.command.c === 'sendPipelineRequest' ||
-      this.command.c === 'sendHttpRequest' ||
-      this.command.c === 'getWebStorageEntry'
-    ) {
-      return;
+  buildCommand() {
+    let command = null;
+    if (this.name) {
+      command = {
+        c: this.name,
+        ...this.params && { p: this.params },
+      };
     }
 
-    const title = `AppCommand %c${this.command.c}`;
-
-    if (this.command.p) {
-      logGroup(title, this.command.p, '#8e44ad');
-    } else logGroup(title, {}, '#8e44ad');
+    return command;
   }
 
   /**
-   * Sends the command to the development API.
-   * @private
+   * Dispatches the command to the app.
+   * The returned promise will not be rejected for now in error cases to avoid the necessity
+   * of refactoring within existing code. But it resolves with FALSE in those cases.
+   * @param {Object} params The command params.
+   * @return {Promise}
    */
-  sendDevCommand() {
-    // Append an optional suffix for special command related endpoints
-    let suffix = '';
-
-    if (this.command.c === 'getWebStorageEntry') {
-      suffix = 'web_storage';
-    } else if (this.command.c === 'sendHttpRequest') {
-      suffix = 'http_request';
+  async dispatch(params) {
+    if (params) {
+      this.setCommandParams(params);
+    }
+
+    const command = this.buildCommand();
+
+    // Only proceed if the command is valid.
+    if (command === null) {
+      logger.error('Command dispatch without command name');
+      return false;
+    }
+
+    let appLibVersion = this.libVersion;
+    let appHasSupport = true;
```

Comment:
Again, please change to a conditional assignment: `const appHasSupport = this.checkLibVersion ? isVersioAtLeat(...) : true;`;